sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -322,9 +332,6 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
             isSimpleExpression(l) && isSimpleExpression(r)) {
             val oldLeft = newLeft
             val oldRight = newRight
-            // Check if the current join is a shuffle join or a broadcast join that
-            // has a shuffle below it
-            val hasShuffle = isProbablyShuffleJoin(left, right, hint)
```

Comment:
Shall we not move this line?
Although move it to line 323 is also correct. We should do it lazily.

Reply:
@beliefer changed it to lazy.

Reply:
In fact, `lazy` doesn't always bring benefits.

Reply:
If we write it here. we could avoid call isProbablyShuffleJoin in most scenarios.

Reply:
I think, it is a tradeoff. If we move the function call down, then it will be called for every pair of leftkey-rightkey. So I feel, it is better to keep it outside the loop.

Reply:
Yes. I agree that is a tradeoff. If we put it outside the loop, any join lead to call `isProbablyShuffleJoin`.
I think we'd better not change it If we don't have a more reasonable reason.